src/v2/System/useFeatureFlag.tsx

```diff
@@ -35,5 +38,44 @@ export function useFeatureVariant(featureName: string): Variant | null {
     return null
   }
 
+  // detremine if we should track the experimentView
+  if (typeof window !== "undefined") {
+    const trackFeatureView = shouldTrack(featureName, variant.name)
+
+    if (trackFeatureView) {
+      trackEvent({
+        action: Schema.ActionType.ExperimentViewed,
+        service: "unleash",
+        experiment_name: featureName,
+        variant_name: variant.name,
+        payload: variant.payload,
+      })
+    }
+  }
```

Comment:
**thought**: maybe we should place this logic in `useEffect`?

Reply:
Ah yeah, nice catch -- that's a must 👍